src/commands/create/create.ts

```diff
@@ -106,42 +103,44 @@ export class CreateCommand implements CommandModule<Argv, CreateArguments> {
     return yargs as Argv<CreateArguments>;
   };
 
-  public handler = async (args: Arguments<CreateArguments>): Promise<void> => {
-    const { awsSecretAccessKey, awsAccessKeyId, pgpassword, pguser, dumpServerToken, tkn, ...restOfArgs } = args;
-    args['dump-server-token-secret'] = undefined;
-    this.logger.debug({ msg: 'starting command execution', command: this.command, args: restOfArgs });
+  const handler = async (args: Arguments<CreateArguments>): Promise<void> => {
+    const { awsSecretAccessKey, awsAccessKeyId, pgpassword, pguser, dumpServerHeaders, ...restOfArgs } = args;
+
+    logger.debug({ msg: 'starting command execution', command: command, args: restOfArgs });
 
     const { stateBucketName, includeState, dumpNameFormat, s3BucketName, s3Acl, dumpServerEndpoint } = restOfArgs;
 
     let isS3Locked = false;
 
+    const manager = dependencyContainer.resolve<CreateManager>(CREATE_MANAGER_FACTORY);
+
     try {
-      if (stateBucketName) {
-        await this.manager.lockS3(stateBucketName);
+      if (stateBucketName !== undefined) {
+        await manager.lockS3(stateBucketName);
         isS3Locked = true;
       }
 
       const dumpMetadataOptions: DumpMetadataOptions = { stateBucketName, includeState, dumpNameFormat };
-      const dumpMetadata = await this.manager.buildDumpMetadata(dumpMetadataOptions, s3BucketName);
+      const dumpMetadata = await manager.buildDumpMetadata(dumpMetadataOptions, s3BucketName);
 
       const pgDumpName = `${dumpMetadata.name}.${PG_DUMP_FILE_FORMAT}`;
-      const pgDumpPath = await this.manager.createPgDump(pgDumpName);
+      const pgDumpPath = await manager.createPgDump(pgDumpName);
 
       if (isS3Locked) {
-        await this.manager.unlockS3(stateBucketName);
+        await manager.unlockS3(stateBucketName as string);
```

Comment:
add a comment explaining why the casting is correct

Reply:
done.